index.js

```diff
@@ -141,44 +188,66 @@ const line = []
 
 
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 1", "color: red")
+console.log("%cScope & Closures - Question 1", "color: red")
 
-// console.log(takeATicketNumber(line))
+let i = 0
+function takeATicketNumber(line) {
+  i += 1
+  line.push(i)
+
+  return `Welcome, you are number ${i} in line.`
+}
+
+console.log(takeATicketNumber(line))
 // // => `Welcome. You are ticket number 1`
 
-// console.log(takeATicketNumber(line))
+console.log(takeATicketNumber(line))
 // // => `Welcome. You are ticket number 2`
 
-// console.log(nowServing(line))
+console.log(nowServing(line))
 // // => `Currently serving 1.`
 
-// console.log(nowServing(line))
-// // => `Currently serving 2.`
+console.log(nowServing(line))
+// => `Currently serving 2.`
 
-// console.log(takeATicketNumber(line))
+console.log(takeATicketNumber(line))
 // // => `Welcome. You are ticket number 3`
 
-// console.log("%c----------", "color: red") 
+console.log("%c----------", "color: red") 
 
 // ***** Scope & Closures - Question 2 *****
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 2", "color: red")
+function ticketNumberGeneratorFunc() {
+  let i = 0
+function takeATicketNumber(line) {
+  i += 1
+  line.push(i)
 
-// const newLine = []
-// const takeATicketNumberFunc = ticketNumberGeneratorFunc()
-// console.log(takeATicketNumberFunc(newLine))
-// // => `Welcome. You are ticket number 1`
+  return `Welcome, you are number ${i} in line.`
+}
+return takeATicketNumber
+}
```

Comment:
good. I would prefer you to write and return an anonymous function within this function

```
function ticketNumberGeneratorFunc() {
  let number = 0;
  return (function (line) {
    number += 1;
    line.push(number);
    return `Welcome. You are ticket number ${number}`;
  })
}
```